ext/civicrm_admin_ui/Civi/Api4/Action/CustomGroup/GetSearchKit.php

```diff
@@ -0,0 +1,233 @@
+    $displayName = $group['entity_name'] . '_Tab';
+    $displayLabel = $group['title'];
+    $description = E::ts('Tab display for %1', [1 => $group['title']]);
+
+    return [
+      'name' => "SavedSearch_{$searchName}_SearchDisplay_{$displayName}",
+      'entity' => 'SearchDisplay',
+      'cleanup' => 'unused',
+      'update' => 'unmodified',
+      'params' => [
+        'version' => 4,
+        'values' => [
+          'name' => $displayName,
+          'label' => $displayLabel,
+          'saved_search_id.name' => $searchName,
+          'type' => $displayType,
+          'settings' => [
+            'placeholder' => 5,
+            'columns' => $columns,
+            'pager' => [
+              'show_count' => TRUE,
+              'expose_limit' => TRUE,
+              'hide_single' => TRUE,
+            ],
+            'headerCount' => TRUE,
+            // only for table but harmless otherwise
+            'actions' => TRUE,
+            'classes' => ['table', 'table-striped'],
+            'actions_display_mode' => 'menu',
+            // only for grid but harmless otherwise
+            'colno' => '3',
+            'toolbar' => [
+              [
+                'action' => 'add',
+                'entity' => $group['entity_name'],
+                'text' => E::ts('Add %1', [1 => $group['title']]),
+                'icon' => 'fa-plus',
+                'style' => 'default',
+                'target' => 'crm-popup',
+                'join' => '',
+                'task' => '',
+                //'condition' => ['check user permission', TRUE],
+              ],
+            ],
+          ],
+        ],
+        'match' => [
+          'saved_search_id',
+          'name',
+        ],
+      ],
+    ];
+  }
+
+  protected function getColumnForField($field) {
+    $key = $field['name'];
+    if ($field['option_group_id']) {
+      $key .= ':label';
+    }
+    // TODO: for entity ref columns, we would like to use
+    // a display-friendly column from the joined entity
+    // but
+    // a) we will need to determine the column based on
+    //    the entity (e.g. display_name for contact)
+    // b) we will need to add it to the SavedSearch as well
+    return [
+      'type' => 'field',
+      'key' => $key,
+      'label' => $field['label'],
+      'sortable' => TRUE,
+      'break' => TRUE,
+    ];
+  }
+
+  protected function getButtonColumn($group) {
+    $groupName = $group['name'];
+    $entityName = $group['entity_name'];
+    return [
+      'size' => 'btn-xs',
+      'links' => [
+        [
+          'entity' => $entityName,
+          'action' => 'view',
+          'target' => 'crm-popup',
+          'icon' => 'fa-eye',
+          'text' => E::ts('View'),
+          'style' => 'default',
+        ],
+        [
+          'entity' => $entityName,
+          'action' => 'update',
+          'target' => 'crm-popup',
+          'icon' => 'fa-pencil',
+          'text' => E::ts('Edit'),
+          'style' => 'warning',
```

Comment:
This isn't the class normally associated with the "Edit" button.

Reply:
Do we have a convention for that?

Delete <=> `danger`  :heavy_check_mark: easy

Then view and edit need to go to two of `default`, `secondary`, `success`, `info`, `warning`?

Reply:
BS3 doesn't really provide any classes to distinguish view from edit. Thus our de-facto so far has been to use `default` for both. Something to bring up with the theming gurus like @artfulrobot or @vingle

Reply:
Have changed to 'default' for now. If it's in line with everywhere else then maybe at some later point we can add some distinction across the board.

Reply:
Yeh there's no universally obvious or agreed match-up between Civi's Submit/Edit/Preview/Cancel/Delete buttons and Bootstrap's default/success/info/warning/danger - tho it would be nice if there was.